src/views/News/index.js

```diff
@@ -134,415 +204,640 @@ export default class StudentNews extends Component {
-      if (this.state.loading === true) {
-        content = <GordonLoader />;
-      } else if (this.state.news.length > 0 || this.state.personalUnapprovedNews.length > 0) {
-        content = (
-          <NewsList
-            news={this.state.filteredNews}
-            personalUnapprovedNews={this.state.personalUnapprovedNews}
-            updateSnackbar={this.updateSnackbar}
-            currentUsername={this.state.currentUsername}
-            callFunction={this.callFunction}
-          />
-        );
-      } else {
-        content = (
-          <Typography variant="h4" align="center">
-            No News To Show
-          </Typography>
-        );
-      }
-
-      let submitButton = (
-        <Button
-          variant="contained"
-          color="primary"
-          onClick={this.handleSubmit.bind(this)}
-          disabled={submitButtonDisabled}
-        >
-          Submit
-        </Button>
+  });
+
+  /* Gets status of current network connection for online/offline rendering
+   *  Defaults to online in case of PWA not being possible
+   */
+  const networkStatus = JSON.parse(localStorage.getItem('network-status')) || 'online';
+
+  if (props.authentication) {
+    if (loading === true) {
+      content = <GordonLoader />;
+    } else if (news.length > 0 || personalUnapprovedNews.length > 0) {
+      content = (
+        <NewsList
+          news={filteredNews}
+          personalUnapprovedNews={personalUnapprovedNews}
+          updateSnackbar={updateSnackbar}
+          currentUsername={currentUsername}
+          callFunction={callFunction}
+        />
       );
-      let editButton = (
-        <Button
-          variant="contained"
-          color="primary"
-          onClick={this.handleUpdate.bind(this)}
-          disabled={submitButtonDisabled}
-        >
-          Update
-        </Button>
+    } else {
+      content = (
+        <Typography variant="h4" align="center">
+          No News To Show
+        </Typography>
       );
+    }
 
-      let news;
-      // If the user is online
-      if (
-        networkStatus === 'online' ||
-        (networkStatus === 'offline' && this.props.authentication)
-      ) {
-        news = (
-          <>
-            {/* Button to Create Posting */}
-            <Fab
-              variant="extended"
-              color="primary"
-              onClick={this.handlePostClick.bind(this)}
-              style={styles.fab}
-            >
-              <PostAddIcon />
-              Post Listing
-            </Fab>
+    let submitButton = (
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={handleSubmit}
+        disabled={submitButtonDisabled}
+      >
+        Submit
+      </Button>
+    );
+
+    let editButton = (
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={handleUpdate}
+        disabled={submitButtonDisabled}
+      >
+        Update
+      </Button>
+    );
```

Comment:
We can consolidate these:
```suggestion
    let submitButton = (
      <Button
        variant="contained"
        color="primary"
        onClick={currentlyEditing ? handleUpdate : handleSubmit}
        disabled={submitButtonDisabled}
      >
        {currentlyEditing ? 'Update' : 'Submit'}
      </Button>
    );
```